packages/type-utils/src/predicates.ts

```diff
@@ -123,24 +124,37 @@ export function isTypeUnknownArrayType(
 
 export enum AnyType {
   Any,
+  PromiseAny,
   AnyArray,
   Safe,
 }
 /**
  * @returns `AnyType.Any` if the type is `any`, `AnyType.AnyArray` if the type is `any[]` or `readonly any[]`,
  *          otherwise it returns `AnyType.Safe`.
  */
-export function isAnyOrAnyArrayTypeDiscriminated(
-  node: ts.Node,
+export function discriminateAnyType(
+  type: ts.Type,
   checker: ts.TypeChecker,
+  program: ts.Program,
 ): AnyType {
-  const type = checker.getTypeAtLocation(node);
   if (isTypeAnyType(type)) {
     return AnyType.Any;
   }
   if (isTypeAnyArrayType(type, checker)) {
     return AnyType.AnyArray;
   }
+  if (isPromiseLike(program, type) && tsutils.isTypeReference(type)) {
+    const awaitedType = type.typeArguments?.[0];
```

Comment:
shouldn't this reuse the logic in `getAwaitedType` (ie by calling it)?

Reply:
@kirkwaiblinger 
 At the time of writing, I thought it was just important to recognize the promise and that recursively getting the type of the actual promise would be bad for performance.
But I haven't actually measured it, but I'm not sure, I'll fix it. 👍